app/src/main/java/com/dessalines/thumbkey/utils/Utils.kt

```diff
@@ -941,3 +942,100 @@ fun getLocalCurrency(): String? {
         ).currency?.symbol
     }
 }
+
+fun makeVariant(
+    source: KeyboardC,
+    variant: KeyboardC,
+): KeyboardC {
+    // Returns a variant of a layer created by replacing source keys and swipes
+    // with any non-dummy equivalents in the variant.
+
+    // Assuming both source and variant have the same structure (same number of rows and columns)
```

Comment:
I think it would be prudent to have some check for source and variant actually having the same structure, although I'm not sure how to enforce that at compile time

Reply:
in the works